### PR TITLE
Modernize argument passing in numeric.h

### PR DESCRIPTION
**Pull request by liangjs**

### CODE REVIEW

File: number_theory/numeric.h [hunk 1]
```diff
@@ -55,7 +55,7 @@ std::pair<T, T> exgcd(T a, T b) {
 template <typename T,
           typename U,
           std::enable_if_t<std::numeric_limits<U>::is_integer, bool> = true>
-T pow(const T &base, U exponent) {
+T pow(T base, const U &exponent) {
```
Comment (umutoztunc) [inline number_theory/numeric.h:58]: Why are we switching `base` to pass-by-value?

**Review by umutoztunc — COMMENTED** [review]
LGTM. I just have a single question about `pow` change.

File: number_theory/numeric.h (hunk 1 above)
Comment (liangjs) [reply number_theory/numeric.h:58]: Notice that base is only used once in this function. If the user pass a rvalue to pow, we can achieve zero-copy.

Comment (umutoztunc) [reply number_theory/numeric.h:58]: I see. If we use const reference we still have to copy once during make_pair. So, const ref is 1 copy all the time whereas pass-by-value is 0 copy in the bestcase and 1 copy in the worstcase. Nice catch!

**Review by umutoztunc — APPROVED** [review]

**Pull request merged**